src/Gui/ToolBarManager.cpp

```diff
@@ -385,3 +387,27 @@ QList<QToolBar*> ToolBarManager::toolBars() const
 
     return tb;
 }
+
+void ToolBarManager::setToolbarVisibility(bool show, const QList<QString>& names) {
+    for (size_t i = 0; i < names.size(); i++) {
+        setToolbarVisibility(show, names[i]);
+    }

```

Comment:
Untested, but this looks to me like it should be a range-based for loop:
```suggestion
    for (const auto &name : names) {
        setToolbarVisibility(show, name);
    }

```

Reply:
I'm not sure to understand the difference. Of course I can fix it here, but should I always use this kind of for loop instead of the more 'traditional' one?

Reply:
Short answer: yes, always use range-based `for` loops. If you want more info, the C++ Core Guidelines are a good place to start: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-for-range

Reply:
Ok thanks. I'll have a look if there's any other loops to change.
Is there any other issue ?

Reply:
I'm waiting for my compile to finish, I'll let you know shortly.

Reply:
> Of course I can fix it here, but should I always use this kind of for loop instead of the more 'traditional' one?

Whenever possible use the range-based for loop as this is more efficient than the traditional way. If you write
`for (auto it = container.begin(); it != container.end(); ++it)` then for each iteration step the function end() must be called. The range-based for loop is syntactic sugar where the compiler creates code to call end() only once.
For more details have a look at: http://antonym.org/2014/02/c-plus-plus-11-range-based-for-loops.html

Hint: Sometimes you need two variables when iterating over a container, the iterator variable and an index. Thanks to boost you can also use the range-based for loop (https://www.fluentcpp.com/2018/10/26/how-to-access-the-index-of-the-current-element-in-a-modern-for-loop/):
```
    for (auto const& it : container | boost::adaptors::indexed(0)) {
        it.value(); // the element of the container
        it.index(); // the index variable
    }
```

Reply:
Thanks for the details. I was wondering about the index, I'll make that down somewhere.